### DISCUSSION TITLE
⏳ 🔗 Allow to set a Vertical Layout for the Mega Menu widget

### DISCUSSION THREAD

**Discussion by danbalaban** [elementor/elementor#22051] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Not having a vertical layout pushes again the use of WordPress menu editor which is not ideal in transferring layouts from site to site. The menu items created in WordPress Menu won't transfer so doing a pure Elementor site to site copy is not ideal.

Think of a scenario with footer columns using vertical navigation! The only self contained solution is using tabs (not ideal) while the WordPress Menu widget relies on items created through the Menu editor (not ideal too).

### Describe the solution you'd like

I would love if you guys would consider **extending the Mega menu** with the ability to also set a **vertical layout for single level menu items**.  This would truly make the process of cross site copy paste complete in a way because the menu items would be completely self contained in the Mega Menu widget and possible to copy paste from domains without an additional step of creating menus in WordPress.

### Describe alternatives you've considered

Custom .css.

**Comment by dasignlady** [comment]
+1

Please include VERTICAL option WITH ICONS. The new widget is touted as the most versatile Wordpress menu, but there is only a dropdown menu option which is worthless for desktop. After spending untold hours trying to manually code icons for the regular vertical menu, it still does not work properly with the built-in hover controls when adding padding since the coded icons are still treated as separate items. This is such a basic item. Why is it missing? I have included a screen capture of my CSS icons to show what the vertical menu should achieve.

![Menu Sidebar Page](https://github.com/elementor/elementor/assets/22346932/11ce2034-35bf-405f-b1ee-dcbd458f7f5e)

**Comment by NAROLLES** [comment]
Helo
use this css with choice horizontal
selector .e-n-menu ul{
  display: flex;
  flex-direction: column;
}

**Reply by phkoon** [reply]
That's great, thanks!

**Reply by phkoon** [reply]
Found the first argument (display: flex;) to be unnecessary, as the the Menu widget's display type is already set to "flex" by default. Is there a reason to include it?

**Reply by webgrade82** [reply]
Hello, If I use this code on mobile, submenu items are below the main menu. Is there any solution to make them like a dropdown? Thanks

**Comment by DanielCyran** [comment]
vertical mega menu please

![image](https://github.com/elementor/elementor/assets/96839460/5f693257-6068-4045-af69-89b8fb0b225b)

**Comment by pavelkrivka** [comment]
+1 Please add vertical layout to MegaMenu
